pkg/client/helper.go

```diff
@@ -228,6 +230,12 @@ func TransportFor(config *Config) (http.RoundTripper, error) {
 		if tlsConfig != nil {
 			transport = &http.Transport{
 				TLSClientConfig: tlsConfig,
```

Comment:
It would probably be better for us to copy the attributes we want:

```
&Transport{
  Proxy: http.DefaultTransport.Proxy,
  Dial: http.DefaultTransport.Dial,
  TLSHandshakeTimeout: http.DefaultTransport.TLSHandshakeTimeout,
}
```

We can't copy the transport without sharing its pool which is verboten (i believe)

----- Original Message -----
> > @@ -228,6 +230,12 @@ func TransportFor(config *Config) (http.RoundTripper,
> > error) {
> >  		if tlsConfig != nil {
> >  			transport = &http.Transport{
> >  				TLSClientConfig: tlsConfig,
> 
> From http://golang.org/pkg/net/http/:
> 
> var DefaultTransport RoundTripper = &Transport{
>         Proxy: ProxyFromEnvironment,
>         Dial: (&net.Dialer{
>                 Timeout:   30 * time.Second,
>                 KeepAlive: 30 * time.Second,
>         }).Dial,
>         TLSHandshakeTimeout: 10 * time.Second,
> }
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4278/files#r24562008